physioqc/metrics/utils.py

```diff
@@ -32,24 +33,29 @@ def print_metric_call(metric, args):
     LGR.info(msg)
 
 
-def physio_or_numpy(signal):
+def physio_or_numpy(func):
     """
-    Return data from a peakdet.physio.Physio object or a np.ndarray-like object.
+    Check if the function input is a Physio object or a np.ndarray-like object.
+
+    The "signal" argument must always be the first one.
 
     Parameters
     ----------
-    data : peakdet.physio.Physio, np.ndarray, or list
-        object to get data from
+    func: function to run the wrapper around
 
     Returns
     -------
-    np.ndarray-like object
-        Either a np.ndarray or a list
+    function
     """
-    if hasattr(signal, "history"):
-        signal = signal.data
 
-    return signal
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        # signal must always be args[0]
+        if hasattr(args[0], "history"):
+            args[0] = args[0].data
+        func(*args, **kwargs)
```

Comment:
I've tried to run some of the multimodal metrics on fake data, and the current code return None. Here's explicitly what I tried 
```
from physioqc.metrics import multimodal
import numpy as np

data = np.random.rand(10)
data_std = multimodal.std(data)
print(data_std)
```
And it just print `None`...

When I tried changing L56 to [explicitly return func](https://docs.python.org/3/library/functools.html#functools.wraps) (i.e., `return func(*args, **kwargs)`), it worked !

Reply:
Ah that makes sense. Yes, please push to my branch!

Reply:
Just created a [PR on your branch](https://github.com/smoia/physioqc/pull/1)